### PR TITLE
Add dynamodb to cloud-governance

### PR DESCRIPTION
**Pull request by athiruma**

### CODE REVIEW

File: cloud_governance/common/aws/cloudtrail/CloudTrail.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        try:
+            role_name = role_arn.split('/')[-1]
+            if event_name == 'CreateUser':
+                start_time = self.__iam_client.get_user(UserName=role_name)['User'].get('CreateDate')
+            elif event_name == 'CreateRole':
+                start_time = self.__iam_client.get_role(RoleName=role_name)['Role'].get('CreateDate')
+            end_time = start_time + timedelta(seconds=self.SEARCH_SECONDS)
+            return [start_time, end_time]
+        except:
+            return ['', '']
```
Comment (ebattat) [inline cloud_governance/common/aws/cloudtrail/CloudTrail.py:35]: Add new line at the end, you can configure it auto in Pycharm

File: jenkins/run_policies.py [hunk 2]
```diff
@@ -79,12 +79,12 @@ def get_custodian_policies(type: str = None):
...
 
 # Gitleaks run on github not related to any aws account
 print("run gitleaks")
 region = 'us-east-1'
 policy = 'gitleaks'
-os.system(f"podman run --rm --name cloud-governance -e policy={policy} -e git_access_token={GITHUB_TOKEN} -e git_repo=https://github.com/redhat-performance -e several_repos=yes -e policy_output=s3://{BUCKET_PERF}/{LOGS}/$region -e log_level=INFO quay.io/ebattat/cloud-governance:latest")
+os.system(f"podman run --rm --name cloud-governance -e policy={policy} e AWS_ACCESS_KEY_ID={AWS_ACCESS_KEY_ID_PERF} -e AWS_SECRET_ACCESS_KEY={AWS_SECRET_ACCESS_KEY_PERF} -e git_access_token={GITHUB_TOKEN} -e git_repo=https://github.com/redhat-performance -e several_repos=yes -e policy_output=s3://{BUCKET_PERF}/{LOGS}/$region -e log_level=INFO quay.io/ebattat/cloud-governance:latest")
```
Comment (ebattat) [inline jenkins/run_policies.py:90]: Missing -e in the parameter
Are you sure we need AWS for gitleaks ?

File: tests/unittest/cloud_governance/dynamodb_upload_data/test_upload_cloudtrail_data_to_dynamodb.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        'AWS::EC2::Instance': instance_id,
+        'Username': 'cloud-governance-test',
+        'Resources': [{'ResourceType': 'AWS::EC2::Instance', 'ResourceName': instance_id}]
+    }]
+    dynamo_db = UploadDataToDynamoDb()
+    dynamo_db._create_table_not_exists(primary_key='EventId')
+    dynamo_db._upload_to_dynamo_db_table(data=mock_log_data)
+    assert len(tag_resources.non_cluster_update_ec2()) == 1
+
+
```
Comment (ebattat) [inline tests/unittest/cloud_governance/dynamodb_upload_data/test_upload_cloudtrail_data_to_dynamodb.py:83]: We dont need mock here, how its work ?

File: jenkins/run_policies.py (hunk 2 above)
Comment (athiruma) [reply jenkins/run_policies.py:90]: yes, we are uploading data to the AWS bucket.

File: tests/unittest/cloud_governance/dynamodb_upload_data/test_upload_cloudtrail_data_to_dynamodb.py (hunk 3 above)
Comment (athiruma) [reply tests/unittest/cloud_governance/dynamodb_upload_data/test_upload_cloudtrail_data_to_dynamodb.py:83]: It test data is uploaded to dynamodb table.

**Review by ebattat — APPROVED** [review]
/LGTM

**Review by ebattat — APPROVED** [review]

**Pull request merged**